cpp/include/raft/spatial/knn/detail/ivf_flat_search.cuh

```diff
@@ -897,8 +898,8 @@ struct euclidean_dist<Veclen, int8_t, int32_t> {
   __device__ __forceinline__ void operator()(int32_t& acc, int32_t x, int32_t y)
   {
     if constexpr (Veclen > 1) {
-      const auto diff = static_cast<int32_t>(__vabsdiffs4(x, y));
-      acc             = dp4a(diff, diff, acc);
+      const auto diff = __vabsdiffs4(x, y);
+      acc             = dp4a(diff, diff, static_cast<uint32_t>(acc));
```

Comment:
So, do I understand the problem right, that `dp4a.s32.s32` interprets each byte as `int8_t`, whereas `__vabsdiffs4` produces four `uint8_t` values (thus, the outputs above 127 interpreted as negative)? And the solution is to do `dp4a.u32.u32`, which interprets components correctly as `uint8_t`? Then, could you please maybe add a short note here that we really want the unsigned overload of `dp4a`?